packages/cojson-transport-ws/src/BatchedOutgoingMessages.ts

```diff
@@ -0,0 +1,36 @@
+import { SyncMessage } from "cojson";
+
+export const MAX_OUTGOING_MESSAGES_CHUNK_SIZE = 10;
```

Comment:
This seems small and it would be better to do the limit roughly by (serialised) data size. I'd suggest limiting it to 1MB initially (if a single message is bigger, that's fine)

Please may you change it to work like that?

Reply:
> This seems small and it would be better to do the limit roughly by (serialised) data size.

Good idea!

> I'd suggest limiting it to 1MB initially

WebSockets are working on top of TCP so if we go over the [computed window size](https://networklessons.com/cisco/ccie-routing-switching-written/tcp-window-size-scaling) it will take more than one roundtrip to send the data, slowing down the exchange instead of speeding it up.

The window size depends on the connection quality, but I would say that 50Kb or even 25Kb would be a safer boundary.

My goal with a chunk of 10 was 1Kb 😅

Reply:
Interesting - the current max chunk size for binary data / transactions is 100kb, wondering what you think of that regarding TCP window size. But for now let's do 25kb of batched message limit - if an individual message is bigger, it's fine

Reply:
Done!